load-gen/src/main/java/io/smartcat/data/loader/api/RateGenerator.java

```diff
@@ -0,0 +1,6 @@
+package io.smartcat.data.loader.api;
+
+public interface RateGenerator {
+
+    long getRate(long prevInstant, long currentInstant);
```

Comment:
Linear generator can be used to torture a target system, looking for a breaking point :)
Anyhow, x is a time value. Now, it could be absolute time (e.g. unix timestamp in ms) or a relative time (e.g. in ms from the ranger start instant). I wold say it should be relative time.